derive-tag/src/lib.rs

```diff
@@ -0,0 +1,69 @@
+use proc_macro::TokenStream;
+use quote::quote;
+use syn;
+
+#[proc_macro_derive(ToTag)]
+pub fn to_tag_derive(input: TokenStream) -> TokenStream {
+    let ast = syn::parse(input).unwrap();
+    impl_to_tag(&ast)
+}
+
+fn impl_to_tag(ast: &syn::DeriveInput) -> TokenStream {
+    let name = &ast.ident;
+
+    let gen = quote! {
+        impl ToTag for #name {
+            fn to_tag(&self) -> u8 {
+                unsafe { *(self as *const Self as *const u8) }
+            }
+        }
+    };
+
+    gen.into()
+}
+
+#[proc_macro_derive(FromTag)]
+pub fn from_tag_derive(input: TokenStream) -> TokenStream {
+    let ast = syn::parse(input).unwrap();
```

Comment:
If you like the `to_compile_error` [suggestion](https://github.com/hachi-bitto/btc-wallet/pull/12/files#r695147574), then I think this should be changed as well.